apps/language_server/lib/language_server/providers/execute_command/expand_macro.ex

```diff
@@ -0,0 +1,52 @@
+defmodule ElixirLS.LanguageServer.Providers.ExecuteCommand.ExpandMacro do
+  @moduledoc """
+  This module implements a custom command expanding an elixir macro.
+  Returns a formatted source fragment.
+  """
+
+  alias ElixirLS.LanguageServer.Server
+
+  @behaviour ElixirLS.LanguageServer.Providers.ExecuteCommand
+
+  @impl ElixirLS.LanguageServer.Providers.ExecuteCommand
+  def execute("expandMacro", [uri, text, line], state)
+      when is_binary(text) and is_integer(line) do
+    source_file = Server.get_source_file(state, uri)
+    cur_text = source_file.text
+
+    if String.trim(text) != "" do
+      result = ElixirSense.expand_full(cur_text, text, line + 1)
+
+      formatted =
+        for {key, value} <- result,
+            into: %{},
+            do:
+              (
+                key =
+                  key
+                  |> Atom.to_string()
+                  |> Macro.camelize()
+                  |> String.replace("Expand", "expand")
+
+                formatted = value |> Code.format_string!() |> List.to_string()
+
+                {key, formatted <> "\n"}
+              )
```

Comment:
I think using `Map.new/2` here would be more clear:

```suggestion
      formatted =
        ElixirSense.expand_full(cur_text, text, line + 1)
        |> Map.new(fn {key, value} ->
          key =
            key
            |> Atom.to_string()
            |> Macro.camelize()
            |> String.replace("Expand", "expand")
                  
          formatted = value |> Code.format_string!() |> List.to_string()
          {key, formatted <> "\n"}
        end)
```

In particular I don't like the long multi-line `do:` clause. But generally because we're not using the for comprehensions filtering it doesn't feel like a good fit here.